gateway/api/guardrails/guardrails.go

```diff
@@ -40,22 +40,32 @@ func Post(c *gin.Context) {
 		CreatedAt:   time.Now().UTC(),
 		UpdatedAt:   time.Now().UTC(),
 	}
-	err := models.CreateGuardRailRules(rule)
+
+	log.Infof("Creating guardrail %s with connections: %v", rule.Name, req.ConnectionIDs)
+
+	// Create guardrail and associate connections in a single transaction
+	err := models.UpsertGuardRailRuleWithConnections(rule, req.ConnectionIDs, true)
+
 	switch err {
 	case models.ErrAlreadyExists:
 		c.JSON(http.StatusConflict, gin.H{"message": err.Error()})
+		return
 	case nil:
-		c.JSON(http.StatusOK, &openapi.GuardRailRuleResponse{
-			ID:          rule.ID,
-			Name:        rule.Name,
-			Description: rule.Description,
-			Input:       rule.Input,
-			Output:      rule.Output,
-			CreatedAt:   rule.CreatedAt,
-			UpdatedAt:   rule.UpdatedAt,
+		log.Infof("Guardrail %s created successfully with %d connection IDs",
```

Comment:
Remove this log statement please, it's not necessary since we're already auditing this endpoint in a middleware